src/util/symbol_generator.cpp

```diff
@@ -0,0 +1,22 @@
+
+#include <util/i2string.h>
+#include <util/symbol_generator.h>
+
+symbolt &symbol_generator::new_symbol(
+  contextt &context,
+  const typet &type,
+  const std::string &name_prefix)
+{
+  symbolt new_symbol;
+  symbolt *symbol_ptr;
+
+  do
+  {
+    new_symbol.name = name_prefix + i2string(++counter);
+    new_symbol.id = prefix + id2string(new_symbol.name);
+    new_symbol.lvalue = true;
+    new_symbol.type = type;
+  } while(context.move(new_symbol, symbol_ptr));
+
+  return *symbol_ptr;
```

Comment:
Is there any chance for this to cause a segmentation fault by `symbol_ptr` being `NULL`? Could you just guard it with an assertion?

Reply:
There is not, see https://github.com/esbmc/esbmc/blob/35f03906d83bb9ae56d9f97223142e10f54bf287/src/util/context.cpp#L26

Reply:
Ow nice